src/main/java/seedu/us/among/model/endpoint/Endpoint.java

```diff
@@ -25,45 +25,44 @@
     /**
      * Every field must be present and not null.
      */
-    public Endpoint(Name name, Address address, Set<Tag> tags) {
-        requireAllNonNull(name, address, tags);
-        this.name = name;
+    public Endpoint(Method method, Address address, Set<Tag> tags) {
+        requireAllNonNull(method, address, tags);
+        this.method = method;
         this.address = address;
         this.tags.addAll(tags);
     }
 
-    public Name getName() {
-        return name;
+    public Method getMethod() {
+        return method;
     }
 
     public Address getAddress() {
         return address;
     }
 
     /**
-     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
-     * if modification is attempted.
+     * Returns an immutable tag set, which throws
+     * {@code UnsupportedOperationException} if modification is attempted.
      */
     public Set<Tag> getTags() {
         return Collections.unmodifiableSet(tags);
     }
 
     /**
-     * Returns true if both persons have the same name.
-     * This defines a weaker notion of equality between two persons.
+     * Returns true if both methods have the same method and address. This defines a weaker
+     * notion of equality between two methods.
      */
     public boolean isSameEndpoint(Endpoint otherEndpoint) {
         if (otherEndpoint == this) {
             return true;
         }
-
-        return otherEndpoint != null
-                && otherEndpoint.getName().equals(getName());
+        return otherEndpoint != null && otherEndpoint.getMethod().equals(getMethod())
+                && otherEndpoint.getAddress().equals(getAddress());
```

Comment:
Might want to be careful of this. Our application might not need this additional check as even having the same method/url may not mean they are the same if the data being sent is different (i.e. data in request body in a POST request)